opal/corelib/runtime.js

```diff
@@ -1817,9 +1817,16 @@
     }
 
     new_body = function() {
-      body.$$p = new_body.$$p;
-      new_body.$$p = null;
-      return body.apply(this, arguments);
+      var block = new_body.$$p;
+      var args = new Array(arguments.length);
+
+      for(var index = 0; index < arguments.length; index++) {
```

Comment:
@jgaskins can you cache the length argument too, e.g. `for (var i = 0, ii = arguments.length; i < ii; i++)`?

Or otherwise do we have a preferred way of dealing with for loops? 
I think this is worth standardizing to let the special cases stand out (e.g. descending index loops). /cc @meh